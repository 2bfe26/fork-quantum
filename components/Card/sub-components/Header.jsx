import styled from 'styled-components';

const Header = styled.header`
  display: flex;
  padding: 20px 20px 0;

  > * {
    margin-right: 18px;
  }
`;

Header.displayName = 'Card.Header';

export default Header;
