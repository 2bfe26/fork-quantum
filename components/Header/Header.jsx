import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider, injectGlobal } from 'styled-components';
import axios from 'axios';

import HeaderBar from './sub-components/HeaderBar';
import PromotionBar from './sub-components/PromotionBar';
import Menu from './sub-components/Menu';

/* eslint no-unused-expressions: ["error", { "allowTaggedTemplates": true }] */
injectGlobal`
  @font-face {
    font-family: cathoIconFontB2c;
    src: url('https://static.catho.com.br/vendor/icon-font/fonts/cathoIconFont/b2c/cathoIconFont.ttf') format('truetype');
  }

  @font-face {
    font-family: cathoIconFontSite;
    src: url('https://static.catho.com.br/vendor/icon-font/fonts/cathoIconFont/site/cathoIconFont.ttf') format('truetype');
  }
`;

class Header extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      showPromotion: false,
    };
  }

  componentDidMount() {
    this.fetchPromotion();
  }

  async fetchPromotion() {
    try {
      const { data } = await axios(`${process.env.APPLICATION_PATH}/api/trials`);
      this.setState({ showPromotion: data });
    } catch (error) {
      process.stderr.write(error);
    }
  }

  render() {
    const {
      data,
      path,
      promotionUrl,
      handleMenuOpen,
      handleLoginOpen,
      menuOpened,
      loginOpened,
    } = this.props;

    return (
      <ThemeProvider theme={data.theme}>
        <header>
          <HeaderBar
            data={data.segments}
            path={path}
          />
          <Menu
            data={data}
            handleMenuOpen={handleMenuOpen}
            handleLoginOpen={handleLoginOpen}
            menuOpened={menuOpened}
            loginOpened={loginOpened}
          />
          {
            this.state.showPromotion &&
            <PromotionBar data={data.promotion} href={promotionUrl}>
              Teste por 30 dias grátis em vez de 7 dias. <span style={{ textDecoration: 'underline' }}>Aproveite essa condição exclusiva!</span>
            </PromotionBar>
          }
        </header>
      </ThemeProvider>
    );
  }
}

Header.propTypes = {
  data: PropTypes.instanceOf(Object),
  path: PropTypes.string.isRequired,
  promotionUrl: PropTypes.string,
  handleMenuOpen: PropTypes.func.isRequired,
  handleLoginOpen: PropTypes.func.isRequired,
  menuOpened: PropTypes.string,
  loginOpened: PropTypes.bool,
};

Header.defaultProps = {
  data: null,
  promotionUrl: 'https://seguro.catho.com.br/cadastro-candidato',
  menuOpened: null,
  loginOpened: false,
};

export default Header;
