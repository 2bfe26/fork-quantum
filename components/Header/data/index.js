import Segments from './segments';
import Promotion from './promotion';
import Menu from './menu';

const Data = {
  theme: {
    primaryColor: '#005da4',
    secondaryColor: '#666666',
    borderColor: '#e3e3e3',
    backgroundColor: '#f5f5f5',
    fontFamily: 'Arial',
    fontSize: '12px',
    paddingTop: '10px',
    paddingBottom: '10px',
    StampsHolderPadding: '10px',
  },
  segments: Segments,
  promotion: Promotion,
  menu: Menu,
};

export default Data;
