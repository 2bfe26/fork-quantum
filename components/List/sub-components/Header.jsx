import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Title = styled.span`
  margin-top: 4px;
  margin-bottom: 4px;
`;

const Header = ({ children }) => <Title>{children}</Title>;

Header.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default Header;
