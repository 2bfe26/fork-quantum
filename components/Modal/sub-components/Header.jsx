import styled from 'styled-components';
import Colors from '../../Colors/deprecated';

const Header = styled.h2`
  border-bottom: 2px solid ${Colors.SECONDARY['50']};
  font-size: 20px;
  font-weight: bold;
  margin: 0;
  padding: 18px 24px;
`;

Header.displayName = 'Modal.Header';

export default Header;
