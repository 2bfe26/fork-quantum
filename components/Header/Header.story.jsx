import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { withKnobs, select } from '@storybook/addon-knobs/react';
import Header from './Header';
import data from './data';

const stories = storiesOf('2. SmartComponents', module);
stories.addDecorator(withKnobs);

const options = {
  '/': 'candidatos',
  'https://www.catho.com.br/empresa/': 'empresas',
  'https://www.catho.com.br/educacao/': 'cursos',
};
const defaultValue = '/';
const groupId = 'GROUP-ID1';

stories.add(
  'Header component',
  withInfo('This is a basic usage of the Header component')(() => (
    <Header
      data={data}
      path={select('Path', options, defaultValue, groupId)}
    />)),
);

Header.displayName = 'Header';
