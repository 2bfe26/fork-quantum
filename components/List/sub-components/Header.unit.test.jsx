import { render } from '@testing-library/react';
import Header from './Header';

describe('<Header />', () => {
  it('Should match the snapshot', () => {
    const { container } = render(<Header>Header</Header>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
