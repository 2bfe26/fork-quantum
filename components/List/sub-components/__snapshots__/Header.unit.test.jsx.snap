// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Should match the snapshot 1`] = `
.c0 {
  margin: 4px 0;
}

<span
  class="c0"
>
  Header
</span>
`;
